File: PythonLabs/Lab6/lab6.py
import math
# Recursive factorial method
def factorial(num):
    if num == 0:
        return 1
    else:
        return num * factorial(num - 1)
        
#print(factorial(5))

# Sin(x) formula
def sine_x():
    n = int(input("Enter an approximation limit (n): "))
    x = int(input("Enter an angle in degrees (x): "))
    x = (x * math.pi) / 180.0 # 180 degrees is pi radians
    res = 0
    for i in range(n+1):
        res += (lambda y : ((-1)**i) * (x**(2*i + 1)) / factorial(2*i + 1))(i)

    return res

#print(sine_x())


# The sum goes to 1/n
ans = 0  # Answer will be added later
def sum_till_one_over_n(n):
    ''' Function takes a positive integer and 
        adds all the numbers (as 1/number) till 1/n,
        and returns nothing for some reasons(?)
    '''
    global ans  # Make ans global
    if n > 0:
        ans += 1 / n
        sum_till_one_over_n(n - 1)


#sum_till_one_over_n(6)
#print(ans)

File: C++Labs/Lab6/lab6.cpp
#include <iostream>
using namespace std;

float sum_till_one_over_n(int n) {
    if (n == 1) {
        return 1.0;
    } else {
        return (1.0 / (float)n) + sum_till_one_over_n(n - 1);
    }
}


float sum_till_one_over_n() {
    int n;
    cout << "Please, enter a positive integer value (n): ";
    cin >> n;
    float ans = 0;
    for(int i = 1; i <= n; i++) {
        ans += (1.0 / (float)i);
    }
    return ans;
}

int main() {
    int n;
    cout << "Please, enter a positive integer value (n): ";
    cin >> n;
    cout << "Question 4 function: " << sum_till_one_over_n(n) << endl;

    float res = sum_till_one_over_n();
    cout << "Question 5 function: " << res << endl;

    return 0;
}